actors/miner/src/expiration_queue.rs

```diff
@@ -363,64 +364,50 @@ impl<'db, BS: Blockstore> ExpirationQueue<'db, BS> {
         // (i.e. 14 for 14-day faults), but if something has gone wrong it's safer not to fail if that's not met.
         let mut sectors_rescheduled = Vec::<&SectorOnChainInfo>::new();
         let mut recovered_power = PowerPair::zero();
+        let mut sectors_by_number = BTreeMap::<u64, &SectorOnChainInfo>::new();
 
-        self.iter_while_mut(|_epoch, expiration_set| {
-            let on_time_sectors: BTreeSet<SectorNumber> = expiration_set
-                .on_time_sectors
-                .bounded_iter(ENTRY_SECTORS_MAX)
-                .context("too many sectors to reschedule")?
-                .map(|i| i as SectorNumber)
-                .collect();
-
-            let early_sectors: BTreeSet<SectorNumber> = expiration_set
-                .early_sectors
-                .bounded_iter(ENTRY_SECTORS_MAX)
-                .context("too many sectors to reschedule")?
-                .map(|i| i as SectorNumber)
-                .collect();
-
-            // This loop could alternatively be done by constructing bitfields and intersecting them, but it's not
-            // clear that would be much faster (O(max(N, M)) vs O(N+M)).
-            // If faults are correlated, the first queue entry likely has them all anyway.
-            // The length of sectors has a maximum of one partition size.
-            for sector in sectors.iter() {
-                let sector_number = sector.sector_number;
-                let power = power_for_sector(sector_size, sector);
-                let mut found = false;
-
-                if on_time_sectors.contains(&sector_number) {
-                    found = true;
-                    // If the sector expires on-time at this epoch, leave it here but change faulty power to active.
-                    // The pledge is already part of the on-time pledge at this entry.
-                    expiration_set.faulty_power -= &power;
-                    expiration_set.active_power += &power;
-                } else if early_sectors.contains(&sector_number) {
-                    found = true;
-                    // If the sector expires early at this epoch, remove it for re-scheduling.
-                    // It's not part of the on-time pledge number here.
-                    expiration_set.early_sectors.unset(sector_number);
-                    expiration_set.faulty_power -= &power;
-                    sectors_rescheduled.push(sector);
-                }
-
-                if found {
-                    recovered_power += &power;
-                    remaining.remove(&sector.sector_number);
-                }
+        for sector in sectors.iter() {
+            let epoch = self.quant.quantize_up(sector.expiration);
+            sectors_group.entry(epoch).or_insert(BitField::new()).set(sector.sector_number);
+            sectors_by_number.insert(sector.sector_number, sector);
+        }
+        for (epoch, sectors_field) in sectors_group {
+            let mut expiration_set = self.may_get(epoch)?;
```

Comment:
This isn't correct. We _do_ still need to iterate over the expiration queue until we have no more sectors to reschedule.

However, we can _hopefully_ avoid iterating over each sector being rescheduled per queue entry. That's where I'm looking for the speedup.